fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -436,15 +452,31 @@ public Date getLastModifiedDate() {
         }
         LOGGER.debug("Could not get last modified date property for node {}", node);
 
-        final Date createdDate = getCreatedDate();
         if (createdDate != null) {
             LOGGER.trace("Using created date for last modified date for node {}", node);
             return createdDate;
         }
 
         return null;
     }
+    private long getTimestamp(final String property, final long created) throws RepositoryException {
+        LOGGER.trace("Using {} date", property);
+        final long timestamp = getProperty(property).getDate().getTimeInMillis();
+        if (timestamp < created && created > 0L && !isFrozenResource()) {
+            LOGGER.trace("Updating {} with later created date", property);
+            getNode().setProperty(property, created);
+            return created;
+        }
+        return timestamp;
+    }
 
+    /**
+     * Set the last-modified date to the current date.
+     * @throws RepositoryException if a repository exception occurs
+     */
+    public void touch() throws RepositoryException {
```

Comment:
I agree that removing the checked exception would be a good thing. @awoods concern about `javax.jcr.AccessDeniedException`s can be addressed by catching those at the point where we're writing data to the JCR nodes and rethrowing them as `org.fcrepo.kernel.api.exceptions.AccessDeniedException`s (rather than higher up in the stack)